cmd/homi/common/utils.go

```diff
@@ -73,6 +78,56 @@ func GenerateKeys(num int) (keys []*ecdsa.PrivateKey, nodekeys []string, addrs [
 	return keys, nodekeys, addrs
 }
 
+func GenerateKeysFromMnemonic(num int, mnemonic, path string) (keys []*ecdsa.PrivateKey, nodekeys []string, addrs []common.Address) {
+	// Ethereum key derivation path: m/44'/60'/0'/0/
+	var (
+		key *bip32.Key
+	)
+
+	for _, level := range strings.Split(path, "/") {
+		if len(level) == 0 {
+			continue
+		}
+
+		if level == "m" {
+			seed := pbkdf2.Key([]byte(mnemonic), []byte("mnemonic"), 2048, 64, sha512.New)
+			key, _ = bip32.NewMasterKey(seed)
+		} else {
+			num := uint64(0)
+			var err error
+			if strings.HasSuffix(level, "'") {
+				num, err = strconv.ParseUint(level[:len(level)-1], 10, 32)
```

Comment:
Did you intend to exclude `'` using `level[:len(level)-1]`? Isn't `level[:len(level)-2]` correct?

Reply:
-1 is correct. See https://go.dev/play/p/1ljsxRzvHKC